components/utils/args.py

```diff
@@ -41,3 +59,328 @@ def add_arguments_sorting(a_arg_parser):
+        "required" : False
+    }
+
+arg_site_owner_uuid = {
+        "arg" : "--site_owner_uuid",
+        "default" : "",
+        "help" : "The UUID representing an organization, the entity capable of owning sites.",
+        "required" : False
+    }
+
+arg_site_auth_code = {
+    "arg" : "--site_auth_code",
+    "default" : "",
+    "help" : "A numeric code used by clients to get access to the site. Must be between 6 and 18 digits.",
+    "required" : True
+}
+
+# Pagination related arguments
+arg_pagination_page_limit = {
+        "arg" : "--page_limit",
+        "default" : "",
+        "help" : "The max size a result list can be prior to being split into pages.",
+        "required" : False
+}
+
+arg_pagination_start = {
+        "arg" : "--start",
+        "default" : "",
+        "help" : "The starting item offset for the return of results.",
+        "required" : False
+}
+
+# Report related arguments
+arg_report_name = {
+        "arg" : "--report_name",
+        "default" : "",
+        "help" : "A name for a report.",
+        "required" : True
+}
+
+arg_report_url = {
+        "arg" : "--report_url",
+        "default" : "",
+        "help" : "The unique location at which the report data can be downloaded from.",
+        "required" : True
+}
+
+arg_report_term = {
+        "arg" : "--report_term",
+        "default" : "longterms",
+        "help" : "The duration partition that the report is stored under. Persistent reports are of term 'longterms'.",
+        "required" : False
+}
+
+arg_report_iso_date_time_start = {
+        "arg" : "--report_iso_date_time_start",
+        "default" : "",
+        "help" : "Start date and time epoch that commences a period of time usually specified when creating a report. ISO format.",
+        "required" : True
+}
+
+arg_report_iso_date_time_end = {
+        "arg" : "--report_iso_date_time_end",
+        "default" : "",
+        "help" : "End date and time epoch that terminates a period of time usually specified when creating a report. ISO format.",
+        "required" : True
+}
+
+arg_report_mask_region_uuid = {
+    "arg" : "--report_mask_region_uuid",
+    "default" : "",
+    "help" : "The RDM UUID representing the region that can be used to optionally bound the physical area that Height Map (AsBuilt) reports are generated over.",
+    "required" : False
+}
+
+arg_report_task_uuid = {
+    "arg" : "--report_task_uuid",
+    "default" : "",
+    "help" : "The RDM UUID representing the task that can be used to optionally filter Height Map (AsBuilt) reports.",
+    "required" : False
+}
+
+arg_report_sequence_instance = {
+    "arg" : "--report_sequence_instance",
+    "default" : "",
+    "help" : "The sequence (sometimes called lift) that can be used to optionally filter Height Map (AsBuilt) reports. For level sequences: index formatted '%08d'; for shift sequences: 'YYYY-MM-DD`T`{startTime}' in site-local time.",
```

Comment:
is this right?  may want to double check with a recent sequence object of shift type. also if this is for filtering an asbuilt report, it should be sequenice id - which I believe is <task_id>:<sequence_order> where order is as you say for lift, but 'HH:MM' for shift:
sprintf(startTimeBuff, "%02d:%02d", int(tod.hours()), int(tod.minutes()));

Reply:
Looks like formatting is correct but parameter nesting may not be. Have raised SHTS-3338.